heat/core/tests/test_signal.py

```diff
@@ -119,3 +119,116 @@ def test_convolve(self):
+
+                conv = ht.convolve2d(signal, dis_kernel_odd, mode=mode)
+                gathered = manipulations.resplit(conv, axis=None)
+                self.assertTrue(
+                    ht.equal(full_odd[i : len(full_odd) - i, i : len(full_odd) - i], gathered)
+                )
+
+                # different data types
+                conv = ht.convolve2d(dis_signal.astype(ht.float), kernel_odd)
+                gathered = manipulations.resplit(conv, axis=None)
+                self.assertTrue(ht.equal(full_odd.astype(ht.float), gathered))
+
+                conv = ht.convolve2d(dis_signal.astype(ht.float), dis_kernel_odd)
+                gathered = manipulations.resplit(conv, axis=None)
+                self.assertTrue(ht.equal(full_odd.astype(ht.float), gathered))
+
+                conv = ht.convolve2d(signal.astype(ht.float), dis_kernel_odd)
+                gathered = manipulations.resplit(conv, axis=None)
+                self.assertTrue(ht.equal(full_odd.astype(ht.float), gathered))
+
+                # even kernel size
+                # skip mode 'same' for even kernels
+                if mode != "same":
+                    conv = ht.convolve2d(dis_signal, kernel_even, mode=mode)
+                    dis_conv = ht.convolve2d(dis_signal, dis_kernel_even, mode=mode)
+                    gathered = manipulations.resplit(conv, axis=None)
+                    dis_gathered = manipulations.resplit(dis_conv, axis=None)
+
+                    if mode == "full":
+                        self.assertTrue(ht.equal(full_even, gathered))
+                        self.assertTrue(ht.equal(full_even, dis_gathered))
+                    else:
+                        self.assertTrue(ht.equal(full_even[3:-3, 3:-3], gathered))
+                        self.assertTrue(ht.equal(full_even[3:-3, 3:-3], dis_gathered))
+
+                # distributed large signal and kernel
+                np.random.seed(12)
+                np_a = np.random.randint(1000, size=(140, 250))
+                np_b = np.random.randint(1000, size=(39, 17))
+                sc_conv = sig.convolve2d(np_a, np_b, mode=mode)
+
+                a = ht.array(np_a, split=0, dtype=ht.int32)
+                b = ht.array(np_b, split=0, dtype=ht.int32)
+                conv = ht.convolve2d(a, b, mode=mode)
+                self.assert_array_equal(conv, sc_conv)
+
+                a = ht.array(np_a, split=1, dtype=ht.int32)
+                b = ht.array(np_b, split=1, dtype=ht.int32)
+                conv = ht.convolve2d(a, b, mode=mode)
+                self.assert_array_equal(conv, sc_conv)
+
+        # test edge cases
+        # non-distributed signal, size-1 kernel
+        signal = ht.arange(0, 16).reshape(4, 4).astype(ht.int)
+        alt_signal = ht.arange(16).reshape(4, 4).astype(ht.int)
+        # please review and then remove: here we require (for whathever reason): ht.float32 instead of ht.int
+        kernel = ht.ones(1).reshape((1, 1)).astype(ht.float32)
```

Comment:
on my machine this was where the unittests failed if `ht.float32` is replaced by `ht.int`